custom/icds_reports/management/commands/delete_users_for_state.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+from django.core.management.base import BaseCommand
+
+from custom.icds_reports.models.views import AwcLocationMonths
+from corehq.apps.locations.dbaccessors import get_users_by_location_id
+from corehq.apps.locations.models import SQLLocation
+
+
+class Command(BaseCommand):
+
+    def get_location_ids(self, loaction_obj):
+        state_ids = set([location.state_id for location in loaction_obj])
+        district_ids = set([location.district_id for location in loaction_obj])
+        block_ids = set([location.block_id for location in loaction_obj])
+        supervisor_ids = set([location.supervisor_id for location in loaction_obj])
+        awc_ids = set([location.awc_id for location in loaction_obj])
+        return state_ids,district_ids,block_ids, supervisor_ids, awc_ids
+
+    def handle(self, *args, **kwargs):
+        telangana_locations = AwcLocationMonths.objects.filter(state_name='Telangana', aggregation_level=5)
+        state_ids, district_ids, block_ids, supervisor_ids, awc_ids = self.get_location_ids(telangana_locations)
+
+        all_locations = state_ids | district_ids | block_ids | supervisor_ids | awc_ids
+
+        location_users = list()
+        user_submitted_forms = []
+        for loc in all_locations:
+            users_to_delete = get_users_by_location_id('icds-cas', loc).all()
```

Comment:
:+1:

Reply:
hey @esoergel Other things are pretty straight forward and minute. on this I have one question for you. This was returning the active users not the deactivated ones. you have some clean way to do so?

Reply:
There are two user "states" that sound kinda similar:

* active vs inactive (deactivated) - that's the thing that's controlled by the "deactivate" button on the user list page

* normal vs retired (deleted) - that's controlled by the "permanent actions" tab on the user's page.  In the UI it's called "Delete", but in actuality it calls the `.retire()` method, which changes the users `base_doc` so the user doesn't show up in most couch views any more.

I'd expect `get_users_by_location_id` to show inactive users, but not retired users.  Is that what you're seeing?  Did Derek retire users or just deactivate them?

Reply:
This was returning around 17K Users, and Derek just deactivated them. And checked a user from telangana which was deactivated but not reeturned by thsi function

Reply:
lets offline